src/OhmmsSoA/VectorSoaContainer.h

```diff
@@ -114,68 +98,67 @@ struct VectorSoaContainer
   }
 
   /** resize myData
-       * @param n nLocal
-       *
-       * nAllocated is used to ensure no memory leak
-       */
+   * @param n nLocal
+   *
+   * nAllocated is used to ensure no memory leak
+   */
   __forceinline void resize(size_t n)
   {
     static_assert(std::is_same<Element_t, typename Alloc::value_type>::value,
                   "VectorSoaContainer and Alloc data types must agree!");
-    if (n == 0)
-    {
-      free();
-      return;
-    }
-    else if (n == nLocal)
-    {
-      nLocal = n;
-    }
-    else
+    if (isRefAttached())
+      throw std::runtime_error("Resize not allowed on VectorSoaContainer constructed by initialized memory.");
```

Comment:
I wondered about this behavior, I'd rather see it this way as well.

Reply:
We have the guard in Matrix and Vector already, probably I forgot to had a look at the SoA container.